[ko] Update /ko/docs/concepts/workloads/controllers/job.md

Eundms opened the pull request:
<!--
 Hello!

 PLEASE title the FIRST commit appropriately, so that if you squash all
 your commits into one, the combined commit message makes sense.
 For overall help on editing and submitting pull requests, visit:
  https://kubernetes.io/docs/contribute/suggesting-improvements/

 Use the default base branch, “main”, if you're documenting existing
 features in the English localization.

 If you're working on a different localization (not English), see
 https://kubernetes.io/docs/contribute/new-content/overview/#choose-which-git-branch-to-use
 for advice.

 If you're documenting a feature that will be part of a future release, see
 https://kubernetes.io/docs/contribute/new-content/new-features/ for advice.
-->
### Description

<!--
 Remember to ADD A DESCRIPTION and delete this note before submitting
 your pull request. The description should explain what will change,
 and why.
-->
- Update `Update /ko/docs/concepts/workloads/controllers/job.md` into Korean.

### Issue
#52554
<!--
 If this pull request resolves an open issue, please link the issue in the PR
 description so it will automatically close when the PR is merged.

 See the GitHub documentation for more details and other options:

 https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword
-->

Closes: #52554

wonyongg commented on content/ko/docs/concepts/workloads/controllers/job.md:
```diff
@@ -176,29 +189,42 @@ kubectl logs $pods
-잡의 이름은 유효한 [DNS 서브도메인 이름](/ko/docs/concepts/overview/working-with-objects/names/#dns-서브도메인-이름)이어야 한다.
+다른 모든 쿠버네티스 설정과 마찬가지로, 잡에도 `apiVersion`, `kind`, `metadata` 필드가 필요하다.
+
+컨트롤 플레인이 잡을 위해 새로운 파드를 생성할 때, 잡의 `.metadata.name`은 
```
```suggestion
컨트롤 플레인이 잡에 대한 새로운 파드를 생성할 때, 잡의 `.metadata.name`은 
```
좀 더 자연스럽게 다듬어보았습니다.

wonyongg commented on content/ko/docs/concepts/workloads/controllers/job.md:
```diff
@@ -176,29 +189,42 @@ kubectl logs $pods
+해당 파드의 이름을 지정하는 기준의 일부이다. 잡의 이름은 유효한
+[DNS 서브도메인](/ko/docs/concepts/overview/working-with-objects/names/#dns-서브도메인-이름) 값이어야 
+하지만, 이로 인해 파드 호스트 이름에 예상하지 못한 결과가 발생할 수 있다. 최상의 호환성을 위해서
+이름은 [DNS 레이블](/ko/docs/concepts/overview/working-with-objects/names#dns-label-names)의 
```
```suggestion
이름은 [DNS 레이블](/docs/concepts/overview/working-with-objects/names#dns-label-names)의 
```

내부링크 현지화 가이드라인이 업데이트 되었음으로 문서 내 내부링크에 있는 언어 디렉터리 `/ko` 접두사를 모두 제거해주시길 바랍니다.

참고
[쿠버네티스 한글화 모범사례 - 내부링크 현지화 가이드라인](https://kubernetes.io/ko/docs/contribute/localization_ko_best_practice/#%EB%82%B4%EB%B6%80-%EB%A7%81%ED%81%AC-%ED%98%84%EC%A7%80%ED%99%94-%EA%B0%80%EC%9D%B4%EB%93%9C%EB%9D%BC%EC%9D%B8)

wonyongg commented on content/ko/docs/concepts/workloads/controllers/job.md:
```diff
@@ -176,29 +189,42 @@ kubectl logs $pods
 `.spec.template` 은 `.spec` 의 유일한 필수 필드이다.
 
+`.spec.template` 은 [파드 템플릿](/ko/docs/concepts/workloads/pods/#파드-템플릿)이다. 
+이것은 중접되어 있으며 `apiVersion` 또는 `kind` 가 없다는 것을 제외한다면 
```
```suggestion
이것은 중첩되어 있으며 `apiVersion` 또는 `kind` 가 없다는 것을 제외한다면 
```
typo인 것 같습니다.

wonyongg commented on content/ko/docs/concepts/workloads/controllers/job.md:
```diff
@@ -243,42 +274,49 @@ _작업 큐_ 잡은 `.spec.completions` 를 설정하지 않은 상태로 두고
 - 파드가 정상적으로(gracefully) 종료되면, 중지하는데 시간이 소요된다.
 
-### 완료 모드
+### 완료 모드 {#completion-mode}
```
```suggestion
### 완료 모드
```

영어 원문에 없는 부분인데 추가하신 이유가 있을까요?
한글화 가이드에서는 영문 헤더에 별도의 앵커가 없는 경우 한글화에 맞게 수정하도록 안내하고 있습니다.

특별한 이유가 없다면 헤더에 앵커를 지워주시고,
337, 394Line에 
`[인덱싱된](#완료-모드)`,
`[인덱스](#완료-모드)`로 변경 부탁드립니다.

[참고 - 쿠버네티스 문서 한글화 가이드 내부 링크 관련](https://kubernetes.io/ko/docs/contribute/localization_ko/#:~:text=(%EB%8B%A8%2C%20%EC%95%B5%EC%BB%A4%20%EB%A7%81%ED%81%AC%EC%9D%98%20%EA%B2%BD%EC%9A%B0%20%EA%B8%B0%EC%A1%B4%EA%B3%BC%20%EA%B0%99%EC%9D%B4%20%ED%95%9C%EA%B5%AD%EC%96%B4%EC%97%90%20%EB%A7%9E%EA%B2%8C%20%EC%88%98%EC%A0%95%ED%95%9C%EB%8B%A4.))

wonyongg commented on content/ko/docs/concepts/workloads/controllers/job.md:
```diff
@@ -243,42 +274,49 @@ _작업 큐_ 잡은 `.spec.completions` 를 설정하지 않은 상태로 두고
+  잡은 암시적으로 `NonIndexed` 상태가 된다.
 - `Indexed`: 잡의 파드는 연결된 완료 인덱스를 0에서 `.spec.completions-1` 까지 
-  가져온다. 이 인덱스는 다음의 세 가지 메카니즘으로 얻을 수 있다.
+  가져온다. 이 인덱스는 다음의 네가지 메카니즘으로 얻을 수 있다.
```
```suggestion
  가져온다. 이 인덱스는 다음의 네 가지 메카니즘으로 얻을 수 있다.
```

[수 관형사 + 의존명사]는 띄어쓰는 것으로 알고 있습니다.

wonyongg commented on content/ko/docs/concepts/workloads/controllers/job.md:
```diff
@@ -295,48 +333,275 @@ _작업 큐_ 잡은 `.spec.completions` 를 설정하지 않은 상태로 두고
+인덱스별 백오프를 사용할 때, 인덱스 내에서 불필요한 재시도를 방지할 수 있는 
+새로운 `FailIndex` 잡이 제공된다.
+
+### 파드 실패 정책 {#pod-failure-policy}
```
```suggestion
### 파드 실패 정책
```
이 부분 역시 영어 원문에 없는 앵커이니 한국어로 변경 부탁드립니다.

wonyongg commented on content/ko/docs/concepts/workloads/controllers/job.md:
```diff
@@ -295,48 +333,275 @@ _작업 큐_ 잡은 `.spec.completions` 를 설정하지 않은 상태로 두고
+
+파드 실패 정책의 두 번째 규칙은 `DisruptionTarget` 조건을 가진 실패한 파드에 대해 
+`Ignore` 작업을 지정하여 파드 중단이 
+`.spec.backoffLimit` 재시도 한도에 포함되지 않도록 한다.
```
```suggestion
파드 실패 정책의 두 번째 규칙에서, `DisruptionTarget` 조건을 가진 실패한 파드에 대해 
`Ignore` 작업을 지정하여 파드 중단이 
`.spec.backoffLimit` 재시도 한도에 포함되지 않도록 한다.
```

자연스럽게 다듬어 보았습니다.

wonyongg commented on content/ko/docs/concepts/workloads/controllers/job.md:
```diff
@@ -295,48 +333,275 @@ _작업 큐_ 잡은 `.spec.completions` 를 설정하지 않은 상태로 두고
+API의 몇 가지 요구 사항과 의미는 다음과 같다.
+
+- 잡에 `.spec.podFailurePolicy` 필드를 사용하려면,
+  해당 잡의 파드 템플릿을 `.spec.restartPolicy`을 `Never`로 설정하여 정의해야 한다..
```
```suggestion
  해당 잡의 파드 템플릿을 `.spec.restartPolicy`을 `Never`로 설정하여 정의해야 한다.
```

wonyongg commented on content/ko/docs/concepts/workloads/controllers/job.md:
```diff
@@ -705,126 +1038,127 @@ spec:
-`.spec.podFailurePolicy` 필드로 정의되는 파드 실패 정책은, 클러스터가
-컨테이너 종료 코드와 파드 상태를 기반으로 파드의 실패를 
-처리하도록 활성화한다.
+### 탄력적 인덱싱 잡
```
```suggestion
### 탄력적(Elastic) 인덱싱 잡
```
`엘라스틱`이란 용어가 그대로 쓰일 때 더 직관적인 경우가 있지 않나 싶어 영문 병기 표현을 제안드려봅니다.

wonyongg reviewed (commented):
@Eundms 

꽤나 큰 사이즈라 고생하셨을 것 같습니다. 정말 수고 많으셨습니다.
해당 PR 이후 새롭게 추가된 한글화 가이드 규칙에 대한 내용을 위주로 리뷰했습니다.

쿠버네티스 문서 한글화 가이드, 한글화 모범 사례 - 내부 링크 현지화
```
쿠버네티스 문서 한글화 가이드
신규 문서 작성 : 영어 원문의 내부 링크 경로를 변경 없이 유지한다.
(단, 앵커 링크의 경우 기존과 같이 한국어에 맞게 수정한다.)
기존 문서 갱신 : 과거에 수동으로 추가한 /ko 접두사가 남아있는 경우 이를 제거한다.
```
```
한글화 모범 사례
원문의 헤더 제목에 별도의 영문 앵커 링크가 지정되어 있는 경우, 
해당 앵커의 참조 링크 구문에서도 기존 영문 표기를 유지한다. 
헤더 제목에 별도 명시된 영문 앵커가 없고 제목이 한글로 변환되었다면, 
헤더 제목에 따라 자동 생성되는 한국어 앵커로 참조 링크 구문도 수정한다.
```
위 규칙에 맞게 수정해야 할 부분이 어느정도 있어 내용 확인 부탁드립니다.

wonyongg commented on content/ko/docs/concepts/workloads/controllers/job.md:
```diff
@@ -210,15 +210,15 @@ kubectl logs jobs/pi
 
 `Never` 또는 `OnFailure` 와 같은 
-[`RestartPolicy`](/ko/docs/concepts/workloads/pods/pod-lifecycle/#재시작-정책)만 허용된다.
+[`RestartPolicy`](/docs/concepts/workloads/pods/pod-lifecycle/#재시작-정책)만 허용된다.
```
영어 원문 파일(`/en/docs/concepts/workloads/pods/pod-lifecycle.md`)에 별도의 영문 앵커가 존재하고,
한국어 파일(`## 컨테이너 재시작 정책 {#restart-policy}`)에도 적용되어 있어 

> \[`RestartPolicy`\](/docs/concepts/workloads/pods/pod-lifecycle/#restart-policy)만 허용된다.

로 수정 부탁드립니다.

ianychoi commented on content/ko/docs/concepts/workloads/controllers/job.md:
```diff
@@ -255,34 +286,41 @@ _작업 큐_ 잡은 `.spec.completions` 를 설정하지 않은 상태로 두고
-  [정적 작업 할당을 사용한 병렬 처리를 위해 인덱싱된 잡](/ko/docs/tasks/job/indexed-parallel-processing-static/)을 참고한다.
-  참고로, 드물기는 하지만, 동일한 인덱스에 대해 둘 이상의 파드를 시작할 수
-  있지만, 그 중 하나만 완료 횟수에 포함된다.
+  [정적 작업 할당을 사용한 병렬 처리를 위해 인덱싱된 잡](/docs/tasks/job/indexed-parallel-processing-static/)을 참고한다.
```
-> `정적 작업 할당을 통한 병렬 처리를 위한 색인된 잡`

ianychoi commented on content/ko/docs/concepts/workloads/controllers/job.md:
```diff
@@ -293,50 +331,277 @@ _작업 큐_ 잡은 `.spec.completions` 를 설정하지 않은 상태로 두고
+  - `Count`: 파드를 기본 방식으로 처리해야 함을 나타낸다.
+     `.spec.backoffLimit`에 대한 카운터를 증가시켜야 한다.
+  - `FailIndex`: 실패한 파드의 인덱스 내에서 불필요한 재시도를 방지하려면
+     [인덱스당 백오프 제한](#인덱스-당-백오프-제한)과 함께 이 작업을 사용한다.
```
`인덱스당` -> `인덱스 당`

ianychoi commented on content/ko/docs/concepts/workloads/controllers/job.md:
```diff
@@ -293,50 +331,277 @@ _작업 큐_ 잡은 `.spec.completions` 를 설정하지 않은 상태로 두고
+실패한 작업의 결과를 실수로 손실되지 않도록 방지하는 것이 좋다.
+{{< /note >}}
+
+### 인덱스 당 백오프 제한
```
"인덱스 당" -> "인덱스당"

Please reflect into all the parts on this file.

- Reference: https://news.hada.io/topic?id=8752

ianychoi commented on content/ko/docs/concepts/workloads/controllers/job.md:
```diff
@@ -293,50 +331,277 @@ _작업 큐_ 잡은 `.spec.completions` 를 설정하지 않은 상태로 두고
+재시도(잡을 실패로 처리하기 이전까지) 횟수를 설정한다. 
+
+백오프 제한은 기본적으로 6으로 설정되어 있다. 단,
+[인덱스당 백오프 제한](#인덱스-당-백오프-제한)(인덱싱된 작업에만 해당)이 지정되어 있지 않으면 그렇지 않는다.
```
Where is the original full notation: `.spec.backoffLimit` ? Also, what does "지정되어 있지 않으면 그렇지 않는다." mean? Please translate a full English sentence into Korean as a complete sentence. My suggestion would be like this:

```
`.spec.backoffLimit`은 [인덱스당 백오프 제한](#인덱스당-백오프-제한)(인덱싱된 작업에만 해당)이
지정되어 있지 않은 이상 6을 기본값으로 설정한다.
```

ianychoi commented on content/ko/docs/concepts/workloads/controllers/job.md:
```diff
@@ -1,23 +1,29 @@
-성공 완료에 도달하면, 작업(즉, 잡)이 완료된다.  잡을 삭제하면 잡이 생성한
-파드가 정리된다. 작업을 일시 중지하면 작업이 다시 재개될 때까지 활성 파드가
+잡은 하나 이상의 파드를 생성하고 지정된 수의 파드가 성공적으로 종료될 때까지 계속해서 파드의 실행을 재시도한다.
+파드가 성공적으로 완료되면, 성공적으로 완료된 잡을 추적한다. 지정된 수가
```
"성공적으로 완료된 잡을 추적한다." -> "해당 잡은 성공적으로 완료된 개수를 추적한다". Original sentence `As pods successfully complete, the Job tracks the successful completions.` does not specify jobs as targets to be tracked.

ianychoi commented on content/ko/docs/concepts/workloads/controllers/job.md:
```diff
@@ -255,34 +286,41 @@ _작업 큐_ 잡은 `.spec.completions` 를 설정하지 않은 상태로 두고
 노드에 그대로 유지되지만, 컨테이너는 다시 실행된다.  따라서 프로그램은 로컬에서 재시작될 때의
 케이스를 다루거나 `.spec.template.spec.restartPolicy = "Never"` 로 지정해야 한다.
-더 자세한 정보는 [파드 라이프사이클](/ko/docs/concepts/workloads/pods/pod-lifecycle/#상태-예제)의 `restartPolicy` 를 본다.
+더 자세한 정보는 [파드 라이프사이클](/docs/concepts/workloads/pods/pod-lifecycle/#example-states)의 `restartPolicy` 를 본다.
```
There is no #example-states on the doc. I have finally found a root cause and a PR to English doc has been raised by myself: https://github.com/kubernetes/website/pull/53818

Eundms commented:
> @Eundms Please see my comments. Also, total # lines on English and Korean docs are different - would you help to make the same # lines on Korean and English docs?

@ianychoi Thank you for the review. : ) 

The difference in the number of lines between the English and Korean documents is because the branch was not rebased properly. I have rebased and updated the Korean document to match the English version.

Thanks again for your feedback!

ianychoi reviewed (commented):
@Eundms I am not sure whether you have reflect all the comments I was raising.

For example,

- https://github.com/kubernetes/website/pull/53186#discussion_r2649074495 - Since `#example-states` does not exist, you need to revise this PR.
- I still see "인덱스 당" five times on the revised doc.
- I still cannot match whether you reflected all of @wonyongg's comments, like https://github.com/kubernetes/website/pull/53186/files#r2596181794 and https://github.com/kubernetes/website/pull/53186/files#r2596189113 .

wonyongg commented:
@Eundms 

> 로컬에서 한국어 헤더에 있는 괄호를 유지한채 아래 2가지 케이스를 확인해보았는데 엥커가 올바르게 작동하지 않았습니다.
따라서, 공백 외의 특수문자가 있는 경우 별도로 엥커를 추가하는 게 좋을 것 같습니다.
공백은 하이픈 으로 변경되는 걸로 알고 있습니다.

테스트 감사드립니다.

특수문자가 들어간 헤더의 경우 앵커가 자동으로 생성되지 않는다면 말씀주신 대로 별도의 앵커를 추가하거나, 영어 원문을 앵커로 추가하는 것도 좋다고 생각합니다. 이 부분에 대해서는 다음 회의 때 안건으로 진행해도 좋을 것 같네요.

ianychoi commented:
For the first part of https://github.com/kubernetes/website/pull/53186#pullrequestreview-3614590800 ,

@Eundms I can still see that L323 has `#example-states`.  
What I meant is that it should be linked to https://kubernetes.io/ko/docs/concepts/workloads/pods/pod-lifecycle/#restart-policy.  
Is there any particular reason you believe your current suggestion on L323 is the correct approach?

@wonyongg — This is not about pointing out a mistake.  
I just wanted to ask for understanding from both the reviewer and approver perspectives.  
Reviewers and approvers work together with contributors to ensure high-quality documentation.  
My earlier question — about why a specific individual was directly asked — was meant to suggest that general technical questions like this might be better discussed in broader Slack channels such as **#sig-docs** or **#sig-docs-localizations**.

I also understand that there are still many open PRs, so everyone is likely under a bit of review overload.  
If we each stay a little more attentive, I believe we can continue to build a stronger and more collaborative community.

Eundms commented:
> For the first part of [#53186 (review)](https://github.com/kubernetes/website/pull/53186#pullrequestreview-3614590800) ,
> 
> @Eundms I can still see that L323 has `#example-states`. What I meant is that it should be linked to https://kubernetes.io/ko/docs/concepts/workloads/pods/pod-lifecycle/#restart-policy. Is there any particular reason you believe your current suggestion on L323 is the correct approach?
> 
> @wonyongg — This is not about pointing out a mistake. I just wanted to ask for understanding from both the reviewer and approver perspectives. Reviewers and approvers work together with contributors to ensure high-quality documentation. My earlier question — about why a specific individual was directly asked — was meant to suggest that general technical questions like this might be better discussed in broader Slack channels such as **#sig-docs** or **#sig-docs-localizations**.
> 
> I also understand that there are still many open PRs, so everyone is likely under a bit of review overload. If we each stay a little more attentive, I believe we can continue to build a stronger and more collaborative community.

@ianychoi 
Thank you for clarifying. 
I replaced the `#example-states` anchor with `#restart-policy`, and updated the following items accordingly.

- line 245 : `모든 파드가 성공적으로 종료되면` -> `어느 하나라도 성공적으로 종료되면` 
  - Original Text  :  `when _any_ Pod from the Job terminates with success, no new Pods are created.` 
- line 295 : `Pod` -> `파드`
  - Update Pod to 파드 for translation consistency
- line 323 : `#example-states` -> `#restart-policy`
  -  Replaced `#example-states` with `#restart-policy`
